File: Digital Business Enablement/buy.it-api/src/main/java/br/com/fiap/buy/it/model/Avaliacao.java
package br.com.fiap.buy.it.model;
import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.*;

import java.util.Date;

@Getter
@Setter
@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Entity
@Table(name = "AVALIACAO", uniqueConstraints = {
        @UniqueConstraint(name = "UK_ID_COTACAO", columnNames = "ID_COTACAO")
})
public class Avaliacao {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "SQ_AVALIACAO")
    @SequenceGenerator(name = "SQ_AVALIACAO", sequenceName = "SQ_AVALIACAO", allocationSize = 1)
    @Column(name = "ID_AVALIACAO")
    private Long id;

    @OneToOne
    @JoinColumn(
            name = "ID_COTACAO",
            referencedColumnName = "ID_COTACAO",
            foreignKey = @ForeignKey(name = "FK_COTACAO_AVALIACAO"),
            nullable = false)
    private Cotacao cotacao;

    @Column(name = "DATA_AVALIACAO", nullable = false)
    @Temporal(TemporalType.DATE)
    private Date data;

    @Column(name = "NOTA_ENTREGA_AVALIACAO", nullable = false)
    @Min(1) @Max(5)
    private Integer notaEntrega;

    @Column(name = "NOTA_QUALIDADE_AVALIACAO", nullable = false)
    @Min(1) @Max(5)
    private Integer notaQualidade;

    @Column(name = "NOTA_PRECO_AVALIACAO", nullable = false)
    @Min(1) @Max(5)
    private Integer notaPreco;

    @Column(name = "DESCRICAO_AVALIACAO", length = 400)
    private String descricao;


}

File: Enterprise Application Development/buyit/Models/Avaliacao.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace br.com.fiap.buy.it.model
{
    [Table("AVALIACAO")]
    public class Avaliacao
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Sequence)]
        [Column("ID_AVALIACAO")]
        public long Id { get; set; }

        [Required(ErrorMessage = "O campo cotacao não pode estar vazio.")]
        [ForeignKey("ID_COTACAO")]
        public long CotacaoId { get; set; }
        public Cotacao Cotacao { get; set; }

        [Required(ErrorMessage = "O campo data não pode estar vazio.")]
        [Column("DATA_AVALIACAO", TypeName = "date")]
        public DateTime Data { get; set; }

        [Required(ErrorMessage = "O campo notaEntrega não pode estar vazio.")]
        [Range(1, 5)]
        [Column("NOTA_ENTREGA_AVALIACAO")]
        public int NotaEntrega { get; set; }

        [Required(ErrorMessage = "O campo notaQualidade não pode estar vazio.")]
        [Range(1, 5)]
        [Column("NOTA_QUALIDADE_AVALIACAO")]
        public int NotaQualidade { get; set; }

        [Required(ErrorMessage = "O campo notaPreco não pode estar vazio.")]
        [Range(1, 5)]
        [Column("NOTA_PRECO_AVALIACAO")]
        public int NotaPreco { get; set; }

        [MaxLength(400)]
        [Column("DESCRICAO_AVALIACAO")]
        public string Descricao { get; set; }
    }
}